waspc/waspls/src/Wasp/LSP/Commands/ScaffoldTsSymbol.hs

```diff
@@ -149,7 +157,29 @@ handler request respond = withParsedArgs request respond scaffold
 
         -- NOTE: we modify the file on disk instead of applying an edit through
         -- the LSP client. See "Current Limitations" above.
-        liftIO $ Text.appendFile (SP.fromAbsFile filepath) rendered
+
+        -- Create directory (and parent directories) if it doesn't exist.
+        liftIO $ Dir.createDirectoryIfMissing True $ SP.fromAbsDir $ SP.parent filepath
+
+        -- If the file already exists, we want to add a blank line between the
+        -- file contents and the new text we're writing.
+        needsLeadingNewline <- liftIO $ Dir.doesFileExist (SP.fromAbsFile filepath)
```

Comment:
This is minor, but I don't think this is the right condition? The file could exist and be empty. In that case, we'll start the file with an empty line.